src/optionals/system.c

```diff
@@ -100,6 +100,36 @@ static Value chownNative(DictuVM *vm, int argCount, Value *args) {
 
     return newResultSuccess(vm, EMPTY_VAL);
 }
+
+static Value mkdirTempNative(DictuVM *vm, int argCount, Value *args) {
+    if (argCount == 0) {
+        runtimeError(vm, "mkdirTemp() takes 1 argument (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_STRING(args[0])) {
+        runtimeError(vm, "mkdirTemp() first argument must be a string");
+        return EMPTY_VAL;
+    }
+
+    char *template = AS_CSTRING(args[0]);
+    char *tmpl = {0};
+
+    if (template[0] != '\0') {
+        tmpl = malloc(strlen(template)+1);
+        strcpy(tmpl, template);
+    } else {
+        tmpl = malloc(PATH_MAX);
+        strcpy(tmpl, "XXXXXX");
+    }
+
+    char *tmpDir = mkdtemp(tmpl);
```

Comment:
I think it's worth using the Result type here as mkdtemp can fail. In those situations it would be good to give some indication as to why it failed, we can do that by unwrapping a Result type